src/client/datascience/notebook/kernelProvider.ts

```diff
@@ -60,40 +50,25 @@ export class VSCodeKernelPickerProvider implements INotebookKernelProvider {
     private notebookKernelChangeHandled = new WeakSet<INotebook>();
     private readonly isLocalLaunch: boolean;
     constructor(
-        @inject(KernelSelectionProvider) private readonly kernelSelectionProvider: KernelSelectionProvider,
-        @inject(KernelSelector) private readonly kernelSelector: KernelSelector,
         @inject(IKernelProvider) private readonly kernelProvider: IKernelProvider,
         @inject(IVSCodeNotebook) private readonly notebook: IVSCodeNotebook,
         @inject(INotebookStorageProvider) private readonly storageProvider: INotebookStorageProvider,
         @inject(INotebookProvider) private readonly notebookProvider: INotebookProvider,
         @inject(KernelSwitcher) private readonly kernelSwitcher: KernelSwitcher,
         @inject(IDisposableRegistry) private readonly disposables: IDisposableRegistry,
         @inject(IExtensionContext) private readonly context: IExtensionContext,
-        @inject(IRawNotebookSupportedService) private readonly rawNotebookSupported: IRawNotebookSupportedService,
         @inject(INotebookKernelResolver) private readonly kernelResolver: INotebookKernelResolver,
         @inject(IConfigurationService) private readonly configuration: IConfigurationService,
-        @inject(IJupyterSessionManagerFactory)
-        private readonly jupyterSessionManagerFactory: IJupyterSessionManagerFactory,
         @inject(PreferredRemoteKernelIdProvider)
         private readonly preferredRemoteKernelIdProvider: PreferredRemoteKernelIdProvider,
-        @inject(ICommandManager) private readonly commandManager: ICommandManager
+        @inject(ICommandManager) private readonly commandManager: ICommandManager,
+        @inject(IExtensions) private readonly extensions: IExtensions,
+        @inject(ILocalKernelFinder) private readonly localKernelFinder: ILocalKernelFinder,
+        @inject(IRemoteKernelFinder) private readonly remoteKernelFinder: IRemoteKernelFinder
     ) {
         this.isLocalLaunch = isLocalLaunch(this.configuration);
-
-        this.kernelSelectionProvider.onDidChangeSelections(
-            (e) => {
-                if (e) {
-                    const doc = this.notebook.notebookDocuments.find((d) => d.uri.fsPath === e.fsPath);
-                    if (doc) {
-                        return this._onDidChangeKernels.fire(doc);
-                    }
-                }
-                this._onDidChangeKernels.fire(undefined);
-            },
-            this,
-            disposables
-        );
         this.notebook.onDidChangeActiveNotebookKernel(this.onDidChangeActiveNotebookKernel, this, disposables);
+        this.extensions.onDidChange(this.onDidChangeExtensions, this, disposables);
```

Comment:
Not sure this is a good idea, just becuase user install `xyz extension` we tell VSCode to reload the list of kernels.